frontend/packages/kubevirt-plugin/src/components/modals/boot-order-modal/boot-order-modal.tsx

```diff
@@ -0,0 +1,156 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Modal, Button, Alert } from '@patternfly/react-core';
+import { ModalComponentProps } from '@console/internal/components/factory';
+import { k8sPatch } from '@console/internal/module/k8s';
+import { VMLikeEntityKind, BootableDeviceType } from '../../../types';
+import { getVMLikeModel, getDevices } from '../../../selectors/vm';
+import { PatchBuilder, PatchOperation } from '../../../k8s/utils/patch';
+import { getVMLikePatches } from '../../../k8s/patches/vm-template';
+import { BootOrder } from '../../boot-order';
+import { DeviceType } from '../../../constants';
+
+import './boot-order-modal.scss';
+
+const modalTitle = 'Virtual machine boot order';
+
+export const BootOrderModal = ({
+  vmLikeEntity,
+  isOpen,
+  setOpen,
+  title = modalTitle,
+}: BootOrderModalProps) => {
+  const [devices, setDevices] = React.useState<BootableDeviceType[]>(getDevices(vmLikeEntity));
+  const [initialDeviceList, setInitialDeviceList] = React.useState<BootableDeviceType[]>(
+    getDevices(vmLikeEntity),
+  );
+  const [showUpdatedAlert, setUpdatedAlert] = React.useState<boolean>(false);
+  const [patchError, setPatchError] = React.useState<string>(null);
+
+  const onReload = React.useCallback(() => {
+    const updatedDevices = getDevices(vmLikeEntity);
+
+    setInitialDeviceList(updatedDevices);
+    setDevices(updatedDevices);
+    setUpdatedAlert(false);
+    setPatchError(null);
+  }, [vmLikeEntity]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  // Inform user on vmLikeEntity.
+  React.useEffect(() => {
+    // Check if bootOrder of current device changed.
+    if (!_.isEqualWith(initialDeviceList, getDevices(vmLikeEntity), _.isEqual)) {
```

Comment:
what is the benefit of using isEqualWith here?

Reply:
changed the logic here to only trigger on boot order changes